Add support for ED25519 signatures, include tip in TransferBuilder

gianinbasler opened the pull request:
Introduce a separate Signature class to support both SR25519 and ED25519 signatures.
Read and write signature from/to byte array with correct type.
Allow setting the tip via TransferBuilder so the encoded unsigned ByteData contains the tip as well.
Add toString() methods to the scale types for easier debugging.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/emeraldpay/polkaj?pullRequest=60) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/emeraldpay/polkaj?pullRequest=60) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/emeraldpay/polkaj?pullRequest=60) it.</sub>

gianinbasler commented:
@splix I added validation for ED25519 signatures and added a test to verify that a known ED25519 signature can be validated.
I also added support for transfer_keep_alive (with accompanying tests) and fixed some tests which were failing due to the Westies support.

splix commented:
Thank you!

gianinbasler commented:
@splix Thank you!
